quickstart/targets.go

```diff
@@ -0,0 +1,79 @@
+package quickstart
+
+import (
+	"fmt"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/huh"
+	"github.com/pkg/errors"
+	"github.com/speakeasy-api/openapi-generation/v2/pkg/generate"
+	"github.com/speakeasy-api/sdk-gen-config/workflow"
+	"github.com/speakeasy-api/speakeasy/charm"
+)
+
+func targetBaseForm(inputWorkflow *workflow.Workflow) (*State, error) {
+	var targetName, targetType, sourceName, outputLocation string
+	if _, err := tea.NewProgram(charm.NewForm(huh.NewForm(
+		huh.NewGroup(
+			huh.NewInput().
+				Title("A name for this target:").
+				Placeholder("unique name across this workflow").
+				Validate(func(s string) error {
+					if _, ok := inputWorkflow.Targets[s]; ok {
+						return fmt.Errorf("a source with the name %s already exists", s)
+					}
+					return nil
+				}).
+				Inline(true).
+				Prompt(" ").
+				Value(&targetName),
+			huh.NewSelect[string]().
+				Title("What target would you like to generate:").
+				Description("Choose from this list of supported generation targets. \n").
+				Options(huh.NewOptions(getSupportedTargets()...)...).
+				Value(&targetType),
+			huh.NewSelect[string]().
+				Title("What source would you like to generate this target from:").
+				Description("Choose from this list of existing sources \n").
+				Options(huh.NewOptions(getSourcesFromWorkflow(inputWorkflow)...)...).
+				Value(&sourceName),
+			huh.NewInput().
+				Title("Provide an output location for your generation target (OPTIONAL):").
+				Inline(true).
+				Prompt(" ").
+				Value(&outputLocation),
+		)),
+		"Let's setup a new target for your workflow.",
+		"A target is a set of workflow instructions and a gen.yaml config that defines what you would like to generate.")).
+		Run(); err != nil {
+		return nil, err
+	}
+
+	target := workflow.Target{
+		Target: targetType,
+		Source: sourceName,
+	}
+
+	if outputLocation != "" {
+		target.Output = &outputLocation
+	}
+
+	if err := target.Validate(generate.GetSupportedLanguages(), inputWorkflow.Sources); err != nil {
+		return nil, errors.Wrap(err, "failed to validate source")
+	}
+
+	inputWorkflow.Targets[targetName] = target
+	// TODO: Should we generate initial gen.yaml files for targets as well.
+
+	addAnotherTarget, err := charm.NewBranchCondition("Would you like to add another target to your workflow file?")
```

Comment:
Not sure if you're already working on this, but after I'm done configuring my target I would expect it to say "Go SDK configured successfully! Generating with `speakeasy run`..." and have it kick off  ageneration run immediately

Reply:
You want to kick off a run or you want to point them towards run

Reply:
I feel pretty strongly that the result of running `quickstart` should be an SDK. It's a bit weird to go through an onboarding flow and not be onboarded by the end of it. Our "time to 200" is when they have an SDK in hand, and IMO quickstart should be optimized for time to 200

Reply:
Discussed we will add this when tristan's work merges